contracts/pcv/PCVDepositAggregator.sol

```diff
@@ -0,0 +1,365 @@
+        }
+
+        int128 difference = int128(newDepositWeight) - int128(pcvDepositInfos[depositAddress].weight);
+        pcvDepositInfos[depositAddress].weight = uint128(newDepositWeight);
+
+        totalWeight = uint128(int128(totalWeight) + difference);
+    }
+
+    function removePCVDeposit(address pcvDeposit) external virtual override onlyGuardianOrGovernor {
+        _removePCVDeposit(address(pcvDeposit));
+    }
+
+    function addPCVDeposit(address newPCVDeposit, uint256 weight) external virtual override onlyGovernor {
+        _addPCVDeposit(address(newPCVDeposit), uint128(weight));
+    }
+
+    function setNewAggregator(address newAggregator) external virtual override onlyGovernor {
+        // Add each pcvDeposit to the new aggregator
+        for (uint i=0; i < pcvDepositAddresses.length(); i++) {
+            address pcvDepositAddress = pcvDepositAddresses.at(i);
+            uint128 pcvDepositWeight = pcvDepositInfos[pcvDepositAddress].weight;
+
+            IPCVDepositAggregator(newAggregator).addPCVDeposit(pcvDepositAddress, pcvDepositWeight);
+        }
+
+        // Set all weights to zero (except for us)
+        for (uint i=0; i < pcvDepositAddresses.length(); i++) {
+            address pcvDepositAddress = pcvDepositAddresses.at(i);
+            pcvDepositInfos[pcvDepositAddress].weight = uint128(0);
+        }
+
+        // Rebalance (withdraws everything to us)
+        _rebalance();
+
+        // Send everything over to the new aggregator
+        IERC20(token).safeTransfer(address(newAggregator), IERC20(token).balanceOf(address(this)));
+
+        // Call rebalance on the new aggregator
+        IPCVDepositAggregator(newAggregator).rebalance();
+
+        // Remove all deposits.
+        for (uint i=0; i < pcvDepositAddresses.length(); i++) {
+            address pcvDepositAddress = pcvDepositAddresses.at(i);
+            _removePCVDeposit(pcvDepositAddress);
+        }
+
+        // Finally, set the new aggregator on the rewards asset manager itself
+        IRewardsAssetManager(rewardsAssetManager).setNewAggregator(address(newAggregator));
+    }
+
+    // ---------- View Functions ---------------
+    function resistantBalanceAndFei() external view virtual override returns (uint256, uint256) {
+        return (balance(), 0);
+    }
+
+    function balanceReportedIn() external view virtual override returns (address) {
+        return token;
+    }
+
+    function balance() public view virtual override returns (uint256) {
+        return getTotalBalance();
+    }
+
+    function percentHeld(address pcvDeposit, uint256 depositAmount) external view virtual override returns(Decimal.D256 memory) {
+        uint totalBalanceWithTheoreticalDeposit = getTotalBalance() + depositAmount;
+        uint targetBalanceWithTheoreticalDeposit = IPCVDeposit(pcvDeposit).balance() + depositAmount;
+
+        return Decimal.ratio(targetBalanceWithTheoreticalDeposit, totalBalanceWithTheoreticalDeposit);
```

Comment:
Whats the difference between theoretical and hypothetical

Reply:
Nothing? We could use either term.